src/rust/ide/src/controller/searcher.rs

```diff
@@ -453,6 +458,7 @@ pub struct Searcher {
     database         : Rc<model::SuggestionDatabase>,
     language_server  : Rc<language_server::Connection>,
     parser           : Parser,
+    ide              : controller::Ide,
```

Comment:
Should we keep `project_name` and `parser` as separate fields, if both of them could be accessible from IDE controller?